Source/Moya+Internal.swift

```diff
@@ -12,25 +12,32 @@ public extension MoyaProvider {
         let stubBehavior = self.stubClosure(target)
         let cancellableToken = CancellableWrapper()
 
+        // Allow plugins to modify response
+        let pluginsWithCompletion: Moya.Completion = { result in
+            var result = result
+            self.plugins.forEach { result = $0.processResponse(result, target: target) }
```

Comment:
This _could_ be done with `reduce`, like:

```swift
let result = self.plugins.reduce(result, { resultSoFar, plugin in
  return plugin.processResponse(resultSoFar, target: target)
})
```

Or something similar. Not sure which reads better. Thoughts?